apps/openmw/mwworld/worldimp.cpp

```diff
@@ -1185,10 +1185,10 @@ namespace MWWorld
                 addContainerScripts (getPlayerPtr(), newCell);
                 newPtr = getPlayerPtr();
             }
-            else if (currCell)
+            else if (currCell && newCell)
```

Comment:
I do not understand this set of changes. They seem to change the program logic quite a bit. Also, the error messages don't make sense. There is no meaningful concept of "empty cells".

Reply:
In this block we call the "moveTo" method by using the "->" syntax. If currCell or newCell is nullptr here, there will be a crash. In theory, currCell could be nullptr only for player during new game start. Instead error logging we can throw an exception here to provide a more informative message.

Reply:
Yes, I get that part. What I do not get is your change. The problem is dealing with newCell==nullptr. The old code didn't handle that properly. But the new one doesn't either (see line 1173).
IIRC calling this function with a nullptr would be an error. If that is the case (and if we really want to take care of it), the easiest way to handle the issue would be a check at the beginning of the function that indeed throws an exception.

And if I interpret the "// currCell == NULL should only happen for player, during initial startup" comment correctly, CurrCell==nullptr can only happen when the moved object is the player, then this can be handled the same way (with an additional isPlayer in the check for the exception).

Reply:
Throw exceptions now.